File: GO/Triangulo_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import (
	"testing"
)

func TestTriangulo(t *testing.T) {
	tables := []struct {
		a    float64
		b    float64
		c    float64
		want string
	}{
		{6.0, 4.0, 2.0, "Area = 10.0"},
		{6.0, 4.0, 2.1, "Perimetro = 12.1"},
	}

	for _, table := range tables {
		got := calcArea(table.a, table.b, table.c)

		if got != table.want {
			t.Errorf("Waiting for this %s but the return was this: %s", table.want, got)
		}
	}
}

File: JS/Triangulo.test.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
 const TEST = require('tape')
 const OBJ = require('./Triangulo.js')
 
 TEST('Starting Triangulo test...', (t) => {
     //Test 1
     t.assert(OBJ.calcArea(6.0, 4.0, 2.0) === "Area = 10.0", "Expect: 6.0, 4.0, 2.0 = 10.00")
     //Test 2
     t.assert(OBJ.calcArea(6.0, 4.0, 2.1) === "Perimetro = 12.1", "Expect: 6.0, 4.0, 2.1 = 12.1")
 
     t.end()
 })
  
